### ISSUE TITLE
Incorrect handling for hidden rows

### ISSUE THREAD

**Issue by sparrowek** [atlassian/react-beautiful-dnd#2013]
I have a scenario where some of the rows in a vertical draggable list are hidden. We use display:none for those. But when the list is larger than a screen a you drag an item from the end toward the top a some point an issue arises. Some rows below the hidden rows start to move as a group and not individually.

A sample attached with item 5 hidden - https://codesandbox.io/s/vertical-list-forked-j5giu?file=/index.js
![dnd](https://user-images.githubusercontent.com/1598273/99184902-ba328880-2746-11eb-8ef3-3847f83d7b13.gif)

**Comment by 2stash** [comment]
@sparrowke
Hello! As of now I ignored anything about hidden items in the list.

I can reproduce the issue exactly as you show in the example. I also copied this code to my computer to see if there was any change to make sure it wasn't an issue with codesandbox, and the code behaved the same.

I assumed that there was something with a list that renders a lot, as the code is written, every time you drag an item all items are rerendered, so that adds up quickly.

So I tried to implement the optimization recommended in the training provided on egghead.io. It adds a simple check so only what needs to get rerendered is rerendered.  [https://egghead.io/lessons/react-optimize-performance-in-react-beautiful-dnd-with-shouldcomponentupdate-and-purecomponent](url)

Check out this updated sandbox. I don't have the issue anymore. Well I gave up dragging around after 30 seconds as I think that is sufficient for a realistic use case. https://codesandbox.io/s/vertical-list-forked-25wyv?file=/index.js

Let me know if this fixes the issue for you.

**Comment by sparrowek** [comment]
Nope, it does not work for me. In your corrected example, when you drag item 29, it stops working when you reach item 16. Item 16 and above start to move as a group.

**Comment by sparrowek** [comment]
From the my observation this issue somehow relates to the viewport scrolling. When I reduce the windows size for example by opening the developer toolbar, it happens earlier.

**Comment by 2stash** [comment]
Ok, after all the scroll training I can get it to break immediately on codesandbox. It seems to happen when scrolling up only, I could not get it to reproduce on just a slow down drag.

Do you have a project where this is happening or are you still just looking for a dnd library so testing it out? The only other thing I was going to add, is that I haven't had this issue when using the library. Below is the project I built and I haven't found any bugs related to this.. 

https://2stash.github.io/algoschedule/

https://github.com/2stash/algoschedule

There are two sortable screens . The first one ("Sort By Difficulty") has 4 column lists, and if I shrink the window, they will still scroll fine. The  second one ("sort By Category"), was my solution to this library not working when your row of columns wraps to new row. In that case it will still be draggable but anything that got flex wrapped to a new row does not drag properly. So I did a work around of having a sortable menu do the ordering of the columns, and then the items in the column sort normally. This one is a long column, but it doesn't break on a slow scroll up.

![scrolling error](https://user-images.githubusercontent.com/60017468/100238370-c14f5880-2ef5-11eb-94f6-b25f62593340.gif)
